spec/ddtrace/error_spec.rb

```diff
@@ -33,12 +33,121 @@
     subject(:error) { described_class.build_from(value) }
 
     context 'with an exception' do
-      let(:value) { ZeroDivisionError.new('divided by 0') }
-
+      let(:value) { begin 1 / 0; rescue => e; e; end }
       it do
         expect(error.type).to eq('ZeroDivisionError')
         expect(error.message).to eq('divided by 0')
-        expect(error.backtrace).to be_empty
+        expect(error.backtrace).to include('error_spec.rb')
+      end
+
+      context 'with a cause' do
+        let(:clazz) do
+          Class.new do
+            def root
+              raise 'root cause'
+            end
+
+            def wrapper
+              root
+            rescue
+              raise 'wrapper layer'
+            end
+
+            def call
+              wrapper
+            rescue => e
+              e
+            end
+          end
+        end
+
+        let(:value) do
+          begin
+            clazz.new.call
+          rescue => e
+            e
+          end
+        end
+
+        it 'reports nested errors' do
+          expect(error.type).to eq('RuntimeError')
+          expect(error.message).to eq('wrapper layer')
+
+          if RUBY_VERSION >= '2.1.0'
+            wrapper_error_message = /error_spec.rb:\d+:in.*wrapper': wrapper layer \(RuntimeError\)/
+            caller_stack = /from.*error_spec.rb:\d+:in `call'/
+            root_error_message = /error_spec.rb:\d+:in.*root': root cause \(RuntimeError\)/
+            wrapper_stack = /from.*error_spec.rb:\d+:in `wrapper'/
+
+            expect(error.backtrace)
+              .to match(/
+                         #{wrapper_error_message}.*
+                         #{caller_stack}.*
+                         #{root_error_message}.*
+                         #{wrapper_stack}.*
+                         #{caller_stack}.*
+                         /mx)
```

Comment:
I'm taking the free pass this time: I think the assertion is not the clearest, but neither it is unreadable. Assertion errors also were legible from my experience developing this feature.

I agree that a separate file would allow for fixed line numbers, but it creates context switch between test files. Here we don't have that context switch, but we do have less legible test assertions. ⚖️